lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -191,6 +166,46 @@ def self.is_firebasecmd_supported?(cmd)
 
         true
       end
+
+      def self.get_token
+        client = Signet::OAuth2::Client.new(
+          token_credential_uri: 'https://oauth2.googleapis.com/token',
+          client_id: FirebaseAppDistributionLoginAction::CLIENT_ID,
+          client_secret: FirebaseAppDistributionLoginAction::CLIENT_SECRET,
+          refresh_token: ENV["FIREBASE_TOKEN"]
+        )
+
+        client.fetch_access_token!
+        return client.access_token
+      rescue Signet::AuthorizationError => error
+        UI.crash!("Wrong value for FIREBASE_TOKEN")
+      end
+
+      def self.validate_app!(params)
+        app_id = (params[:app])
+        token = get_token
+
+        connection = Faraday.new(url: BASE_URL) do |conn|
+          conn.response(:json, parser_options: { symbolize_names: true })
+          conn.response(:raise_error) # raise_error middleware will run before the json middleware
+          conn.adapter(Faraday.default_adapter)
+        end
+        begin
+          response = connection.get("#{PATH}#{app_id}") do |request|
+            request.headers["Authorization"] = "Bearer " + token
+          end
+        rescue Faraday::ResourceNotFound => error
+          UI.user_error!("App Distribution could not find your app #{app_id}. Make sure to onboard your app by pressing the \"Get started\" button on the App Distribution page in the Firebase console: https://console.firebase.google.com/project/#{app_id}/appdistribution")
```

Comment:
This uses the App ID in the URL, when what you really need is the Project ID (Projects can contain many Apps). I now realize that in this case you don't actually have the Project ID, since you were unable to fetch the app information. So please disregard my original comment, sorry about that. `https://console.firebase.google.com/project/_/appdistribution` is fine.